File: OCH_Tao/ABC/A/250.js
const input = (require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n"));
const [H, W] = (input[0].split(" ")).map(x => parseInt(x, 10));
const [R, C] = (input[1].split(" ")).map(x => parseInt(x, 10));
let cnt = 0;
if (R != 1) {
  cnt++;
}
if (R != H) {
  cnt++;
}
if (C != 1) {
  cnt++;
}
if (C != W) {
  cnt++;
}
console.log(cnt);

File: OCH_Tao/ABC/A/250.py
H,W = map(int,input().split())
R,C = map(int,input().split())
cnt = 0
if 0<R-1:
  cnt+=1
if R+1<=H:
  cnt+=1
if 0<C-1:
  cnt+=1
if C+1<=W:
  cnt+=1
print(cnt)